tutorials/google-kubernetes-engine-hyperparameter-search/index.md

```diff
@@ -0,0 +1,266 @@
+        }
+    ],
+    'images': [
+        'gcr.io/$PROJECT_ID/IMAGE_NAME'
+    ]
+}
+```
+
+When a build request is successfully executed on Cloud Registry, a docker image will be created and registered with an identifier such as `gcr.io/$PROJECT_ID/IMAGE_NAME`.  Later we will pass this to Kubernetes Engine so the source code packaged in the docker image can be executed on a cluster.
+
+Note that we need to upload a zipped folder containing the source code to Cloud Storage.  The helpers in this sample take care of that for us.
+
+### [`source`][source]
+
+The `source` folder contains the code that will actually be executed on the cluster.  It contains a simple [Dockerfile][Dockerfile] which contains the instructions for Cloud Registry to build the docker image.
+
+The file [worker.py][worker.py] contains the code that retrieves a pickled `SearchCV` object from Cloud Storage and calls its `fit` method.
+
+```python
+# worker.py
+X = download_uri_and_unpickle(X_uri)
+y = download_uri_and_unpickle(y_uri)
+search = download_and_unpickle(bucket_name, 'worker_0/search.pkl')
+
+search.fit(X, y)
+```
+
+Here `download_uri_and_unpickle` and `download_and_unpickle` are helper methods in [gcs_helper.py][gcs_helper.py].
+
+In the code snippet above we see that the training data are provided as Cloud Storage URIs.  This allows you to upload the data just once and use it for multiple hyperparameter search jobs.
+
+### [`gke_helper.py`][gke_helper.py]
+
+To make it easy to create a cluster on which our jobs will run, the sample code includes helpers to send cluster creation requests to Kubernetes Engine.
+
+```python
+# gke_helper.py
+service = discovery.build('container', 'v1', credentials=credentials)
+create = service.projects().zones().clusters().create(body=body, zone=zone, projectId=project_id).execute()
+```
+
+Here the request `body` specifies the cluster to be created, such as the type and number of nodes.  For example, a cluster of one node with 64 cores might be specified as follows:
+
+```python
+# gke_helper.py
+cluster = {
+    'master_auth': {
+        'username': 'admin'
+    },
+    'name': 'YOUR-CLUSTER-ID',
+    'node_pools': [
+        {
+            'name': 'default-pool',
+            'initial_node_count': 1,
+            'config': {
+                'machine_type': 'n1-standard-64',
+                'oauth_scopes': [
+                    # required scopes here
+                ]
+            }
+        }
+    ]
+}
+
+body = {
+    'cluster': cluster
+}
+```
+
+Some `oauth_scopes` are required in order for the cluster to interact with Cloud Registry (to pull the docker image we created) and Cloud Storage (to get the training data and pickled `SearchCV` object).
+
+### [`kubernetes_helper.py`][kubernetes_helper.py]
+
+The execution of the code we packaged as a docker image on a cluster, we need to submit a `job` to the cluster.  We use Kubernetes's client library for that.
```

Comment:
Suggested edit:
To execute the code we packaged as a docker image on a cluster, we need to submit a `job` to the cluster.  We use Kubernetes's client library for that.